cypress/integration/e2e/table/update-workhours-info.spec.js

```diff
@@ -0,0 +1,285 @@
+
+const nurseInitialWorkHours = { required: 136, actual: 164, overtime: 28 };
+const babysitterInitialWorkHours = { required: 80, actual: 108, overtime: 28 };
+
+const workerTypeToIdx = {
+  nurse: 0,
+  babysitter: 3,
+};
+
+function shiftCell(workerType, shiftIdx) {
+  return cy
+    .get(`[data-cy="${workerType}ShiftsTable"]`)
+    .children()
+    .children()
+    .eq(workerTypeToIdx[workerType])
+    .children()
+    .eq(shiftIdx)
+    .children()
+    .children();
+}
+
+const cellTypeToIdx = {
+  required: 0,
+  actual: 1,
+  overtime: 2,
+};
+
+function hoursInfoCell(workerType, cellType) {
+  return cy
+    .get(`[data-cy="${workerType}SummaryTable"]`)
+    .children()
+    .children()
+    .eq(workerTypeToIdx[workerType])
+    .children()
+    .eq(cellTypeToIdx[cellType]);
+}
+
+function testWorkHoursInfoUpdate(testData) {
+  const cellData = [testData.workerType, testData.shiftIndex];
+
+  if (testData.initialShiftCode) {
+    shiftCell(...cellData).contains(testData.initialShiftCode);
+    shiftCell(...cellData).click();
+  } else {
+    shiftCell(...cellData)
+      .parent()
+      .click();
+  }
+  cy.get('[data-cy="shiftDropdown"]').contains(testData.desiredShiftText).click({ force: true });
+
+  if (testData.desiredShiftCode) {
+    shiftCell(...cellData).contains(testData.desiredShiftCode);
+  } else {
+    shiftCell(...cellData).should("be.empty");
+  }
+
+  hoursInfoCell(testData.workerType, "required").contains(testData.expectedWorkHoursInfo.required);
+  hoursInfoCell(testData.workerType, "actual").contains(testData.expectedWorkHoursInfo.actual);
+  hoursInfoCell(testData.workerType, "overtime").contains(testData.expectedWorkHoursInfo.overtime);
+}
+
+context("Work hours info (summary table)", () => {
+  beforeEach(() => {
+    cy.visit(Cypress.env("baseUrl"));
+    cy.contains("Plik").click();
+    cy.get('[data-cy="file-input"]').attachFile("example.xlsx");
+  });
+
+  // sanity check in case schedule in the docs gets changed and these tests start failing because of it
+  it("Has expected initial values of workHourInfo in example schedule", () => {
+    hoursInfoCell("nurse", "required").contains(nurseInitialWorkHours.required);
+    hoursInfoCell("nurse", "actual").contains(nurseInitialWorkHours.actual);
+    hoursInfoCell("nurse", "overtime").contains(nurseInitialWorkHours.overtime);
+
+    hoursInfoCell("babysitter", "required").contains(babysitterInitialWorkHours.required);
+    hoursInfoCell("babysitter", "actual").contains(babysitterInitialWorkHours.actual);
+    hoursInfoCell("babysitter", "overtime").contains(babysitterInitialWorkHours.overtime);
+  });
+
+  describe("For a nurse", () => {
+    describe("When D for current month weekday", () => {
+      it("Is added, should add 12 to actual and overtime hours and not change required", () => {
+        const data = {
+          workerType: "nurse",
+          shiftIndex: 21,
+          initialShiftCode: null,
+          desiredShiftText: "dzień",
```

Comment:
`desiredShiftText` wyniósłbym do słownika/ enuma gdzieś na górze, bo jest duża szansa na to, że tekst w dropdownie się zmieni, a widzę, że teraz powtarza się w dość dużej ilości miejsc

Reply:
Zgadzam się, przy okazji udało się wywalić trochę loc